### PR TITLE
[SPARK-15232][SQL] Add subquery SQL building tests to LogicalPlanToSQLSuite

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

We currently test subquery SQL building using the `HiveCompatibilitySuite`. The is not desired since SQL building is actually a part of `sql/core` and because we are slowly reducing our dependency on Hive. This PR adds the same tests from the whitelist of `HiveCompatibilitySuite` into `LogicalPlanToSQLSuite`.
## How was this patch tested?

This adds more testcases. Pass the Jenkins tests.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @hvanhovell and @rxin

File: sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala [hunk 1]
```diff
@@ -934,6 +934,159 @@ class LogicalPlanToSQLSuite extends SQLBuilderTest with SQLTestUtils {
...
+        |SELECT subq.key1, z.value
+        |FROM (SELECT x.key as key1, x.value as value1, y.key as key2, y.value as value2
+        |      FROM src1 x JOIN src y ON (x.key = y.key)) subq
+        |JOIN srcpart z ON (subq.key1 = z.key and z.ds='2008-04-08' and z.hr=11)
+        |ORDER BY subq.key1, z.value
+      """.stripMargin,
+      "mapjoin_subquery")
+  }
+
+  test("subq2") {
```
Comment (rxin) [inline sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala]: is there a more descriptive name?

File: sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala [hunk 2]
```diff
@@ -934,6 +934,159 @@ class LogicalPlanToSQLSuite extends SQLBuilderTest with SQLTestUtils {
     }
   }
 
+  test("mapjoin_subquery") {
```
Comment (rxin) [inline sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala]: broadcast join

File: sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala [hunk 3]
```diff
@@ -934,6 +934,159 @@ class LogicalPlanToSQLSuite extends SQLBuilderTest with SQLTestUtils {
...
+        |from src b
+        |group by b.key
+        |having exists (select a.key
+        |               from src a
+        |               where a.value > 'val_9' and a.value = min(b.value))
+      """.stripMargin,
+      "subquery_exists_having_3")
+  }
+
+  test("subquery_notexists") {
```
Comment (rxin) [inline sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala]: subquery_not_exists

Comment (rxin) [inline sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala]: same thing for the following few tests.

File: sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala (hunk 1 above)
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala]: Sure. I actually just use the name from Hive suite for review.
I'll review this.

File: sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala (hunk 2 above)
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala]: Sure!

File: sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala (hunk 3 above)
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala]: Yep. For the naming, I will revise all according to the test context.

**Comment by dongjoon-hyun** [comment]
Hi, @rxin and @hvanhovell .
It's ready for review again.

**Comment by rxin** [comment]
Looks reasonable to me. Merging in master/2.0.

**Comment by dongjoon-hyun** [comment]
Thank you for review and merging, @rxin .

**Pull request closed without merging**